segmentstore/server/src/main/java/io/pravega/segmentstore/server/Metrics.java

```diff
@@ -0,0 +1,332 @@
+
+        public void report(long totalBytes, int generationSpread) {
+            this.totalSize.reportSuccessValue(totalBytes);
+            this.generationSpread.reportSuccessValue(generationSpread);
+        }
+    }
+
+    //endregion
+
+    //region ThreadPool
+
+    /**
+     * SegmentStore ThreadPool metrics.
+     */
+    public final static class ThreadPool implements AutoCloseable {
+        private final OpStatsLogger queueSize;
+        private final OpStatsLogger activeThreads;
+        private final ScheduledExecutorService executor;
+        private final ScheduledFuture<?> reporter;
+
+        public ThreadPool(ScheduledExecutorService executor) {
+            this.executor = Preconditions.checkNotNull(executor, "executor");
+            this.queueSize = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_QUEUE_SIZE);
+            this.activeThreads = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_ACTIVE_THREADS);
+            this.reporter = executor.scheduleWithFixedDelay(this::report, 1000, 1000, TimeUnit.MILLISECONDS);
+        }
+
+        @Override
+        public void close() {
+            this.reporter.cancel(true);
+        }
+
+        private void report() {
+            ExecutorServiceHelpers.Snapshot s = ExecutorServiceHelpers.getSnapshot(this.executor);
+            if (s != null) {
+                this.queueSize.reportSuccessValue(s.getQueueSize());
+                this.activeThreads.reportSuccessValue(s.getActiveThreadCount());
+            }
+        }
+    }
+
+    //endregion
+
+    //region OperationProcessor
+
+    /**
+     * OperationProcessor metrics.
+     */
+    public final static class OperationProcessor {
+        /**
+         * Number of items in the Operation Queue.
+         */
+        private final OpStatsLogger operationQueueSize;
+
+        /**
+         * Number of items out of the Operation Queue, sent to Tier1, but not yet acknowledged.
+         */
+        private final OpStatsLogger operationsInFlight;
+
+        /**
+         * Amount of time an operation spends in the queue, before being picked up.
+         */
+        private final OpStatsLogger operationQueueWaitTime;
+
+        /**
+         * Amount of time the OperationProcessor delays in order to relieve Tier1 in case of back-up.
+         */
+        private final OpStatsLogger operationProcessorDelay;
+
+        /**
+         * Amount of time spent committing an operation after being written to Tier1 (this includes in-memory structures
+         * and Cache).
+         */
+        private final OpStatsLogger operationCommitLatency;
+
+        /**
+         * Container-specific, end-to-end latency of an operation from when it enters the OperationProcessor until it is completed.
+         */
+        private final OpStatsLogger operationLatency;
+
+        /**
+         * Amount of time spent ack-ing operations.
```

Comment:
Acking is from when the operation is first received by the `OperationProcessor` or is it really just the processing of the ack?

Reply:
I removed this metric.